open-craft/taxonomy-sample-data
# feat: Implement taxonomy sample data generation

### Description
Implements generating sample taxonomy data to be used for testing and development. Repeating what is already mentioned in the README:

Running this script will do the following:

1. 2 Test Organizations will be created
1. For each organization created, a sample course is created that is populated with a course export that contains a variety of section/subsections/units and components
1. For each of these created organizations the following taxonomies will be created:
    1. a disabled taxonomy with 10 Tags
    1. an enabled flat taxonomy with 5000 Tags
    1. an enabled hierarchical taxonomy with three levels and 4^x tags per level (4 root tags, each with 16 child tags, each with 64 grandchild tags)
    1. a small enabled taxonomy with 2 levels with 2 Tags each
1. A multi org Taxonomy is created and enabled/used by both orgs
1. (Optional) A 4 level Taxonomy containing data obtained from [Open Canada Taxonomy](https://open.canada.ca/data/en/dataset/6093c709-2a0d-4c23-867e-27987a79212c/resource/0a120b15-9708-4d8a-8af2-2431c4540c0b)
1. (Optional) A 3 level Taxonomy containing data obtained from [LightCast Skills Taxonomy](https://docs.google.com/spreadsheets/d/1DA3JfpBE5Krc0daImuu5Y0nsH93PEfdrWRrEa-sR-6k/edit#gid=1319222368)
1. Once the Taxonomies and their Tags have been created, the script will Tag each of the courses along with all the components they contain with 1 of each Tag from the the above

**Note:** This script is designed to be idempotent. Meaning that the end state is the same every time you run it. So if you make modifications to the sample courses on Studio or the Taxonomy data in the shell and run this script again, it will reset all your changes.

### Related Tickets:
- Closes https://github.com/openedx/modular-learning/issues/107

### Testing Instructions
Similar to what is mentioned in the README:
1. To begin, clone this repo inside a directory so it can be accessed from within the devstack (eg: `/edx/src/`)
1. Set the `TAXONOMY_SAMPLE_PATH` variable inside the `generate.py` file to point to the cloned repo inside your devstack/tutor environment, eg: `/edx/src/taxonomy-sample-data`
1. (Optional) If you would not like to include taxonomy data from real world examples, such as:
    - [Open Canada Taxonomy](https://open.canada.ca/data/en/dataset/6093c709-2a0d-4c23-867e-27987a79212c/resource/0a120b15-9708-4d8a-8af2-2431c4540c0b)
    - [LightCast Skills Taxonomy](https://docs.google.com/spreadsheets/d/1DA3JfpBE5Krc0daImuu5Y0nsH93PEfdrWRrEa-sR-6k/edit#gid=1319222368)

    Then set the following flags to `False` in `generate.py` accordingly:

    ```py
    IMPORT_OPEN_CANADA_TAXONOMY = True
    IMPORT_LIGHTCAST_SKILLS_TAXONOMY = True
    ```

1. To run the script, enter the LMS shell (`make lms-shell`) and run the following command:
    ```sh
    python manage.py cms shell < /path/to/taxonomy-sample-data/generate.py
    ```

4. The script should run to completion, you'll notice it tagging the courses + components in the logs of the script

---
Private-ref: [FAL-3516](https://tasks.opencraft.com/browse/FAL-3516)
